lib/files.js

```diff
@@ -0,0 +1,54 @@
+const { checkIfPathMatches } = require('./gitpatternParser')
+
+/**
+ * process the PR based on files settings
+ *
+ * @return
+ *  JSON object with the properties mergeable and the description if
+ *  `mergeable` is false
+ */
+module.exports = async (pr, context, settings) => {
+  let fileSettings = settings.files
+  if (!fileSettings) {
+    return {
+      mergeable: true,
+      description: 'Okay To Merge'
+    }
+  }
+
+  let patternToMatch = settings.files.pattern
+  // fetch the file list
+  let result = await context.github.pullRequests.getFiles(context.repo({number: pr.number}))
+  let addedFiles = result.data.filter(file => file.status === 'added' || file.status === 'modified')
+  // parse the files to match the pattern
+  const filesToCheck = addedFiles.filter(file => checkIfPathMatches(file.filename, patternToMatch))
+
+  // fetch the file content
+  result = await fetchFileContents(filesToCheck, context)
```

Comment:
Might be cleaner to write the code block above (from L21 to L27) as so:
```js
let filesToCheck = await context.github.pullRequests.getFiles(context.repo({number: pr.number})).data
   .filter(file => file.status === 'added' || file.status === 'modified')
   .filter(file => checkIfPathMatches(file.filename, patternToMatch))

let fileContents = await fetchFileContents(filesToCheck, context)
```

Here's why I think so:
- The variable `result` is quickly overwritten with some other function call result and is too loosely defined. Not obviously apparent at first glance.
- `addedFiles` is very temporary and is not used elsewhere except very subsequently to filter again. It's better not to declare it. The reader will not wonder what else it is used for.

Reply:
done